src/models/hooks/sessionReportPilot.js

```diff
@@ -213,6 +213,76 @@ export const createGoalsForSessionRecipientsIfNecessary = async (sequelize, sess
   }
 };
 
+export const removeGoalsForSessionRecipientsIfNecessary = async (sequelize, sessionReportOrInstance, options) => {
+  const processSessionReport = async (sessionReport) => {
+    let event;
+    let nextSessionRecipients;
+
+    if (sessionReport && sessionReport.data) {
+      const data = (typeof sessionReport.data.val === 'string')
+        ? JSON.parse(sessionReport.data.val)
+        : sessionReport.data;
+
+      event = data.event;
+      nextSessionRecipients = data.recipients;
+    }
+
+    if (!event.id || !sessionReport.id) return;
+
+    nextSessionRecipients = nextSessionRecipients.map((r) => r.value);
+
+    const recipients = await sequelize.models.EventReportPilotGoal.findAll({
+      where: { eventId: event.id, sessionId: sessionReport.id },
+      transaction: options.transaction,
+      raw: true,
+    });
+
+    if (!recipients || !recipients.length) return;
+
+    // Using the EventReportPilotGoal table, find all goals that were created for this recipient (grantId).
+    // eslint-disable-next-line no-restricted-syntax
+    for await (const entry of recipients) {
```

Comment:
nitpick entry = recipient

Reply:
entry is a row in `EventReportPilotGoal`, which looks like `{ eventId, goalId, recipientId, sessionId }` - so not exactly a recipient. 

I struggled with naming this variable for .5s before telling myself "it doesn't matter".

I can rename it to `recipient` if you still think it's better?